### PR TITLE
Improve convert command

### PR DESCRIPTION
**Pull request by Darkflame72**
# Initial Checklist
- [x] Has @tigattack / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [x] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/master/README.md#cog-summaries)?

# Details
**Does this resolve an issue?**
Resolves [this](https://discord.com/channels/184315303323238400/281746926662189056/803963156417478718)

### Features / Fixes
- moves to using GNU Units for conversion etween units

### Breaking Changes
None

## Additional
Any further notes or comments you want to make.

### CODE REVIEW

**Comment by tigattack** [comment]
Whoops, sorry Issy. Just noticed this is still a draft.

I'll let @Darkflame72 request review when they're ready.

**Comment by Darkflame72** [comment]
This now works and is tested and has some slightly prettier error messages. The downside is it relies on GNU utils which must be installed on the system. I would be happy to open a pr to the dockerfile to add it.

**Comment by tigattack** [comment]
Thanks a lot @Darkflame72!

> I would be happy to open a pr to the dockerfile to add it.

If you wouldn't mind, that would be great.

Please sort the imports to satisfy isort when you get a moment. I'll leave the rest to @issy's superior Python knowledge 😄

**Comment by Darkflame72** [comment]
Created the PR for the docker image [here](https://github.com/rHomelab/Red-DiscordBot-Docker/pull/10).

File: convert/convert.py [hunk 1]
```diff
@@ -1,35 +1,66 @@
...
-from pint import UnitRegistry
 from redbot.core import commands
 
 
 class Convert(commands.Cog):
     """Convert related commands."""
 
     def __init__(self):
-        self.__ureg = UnitRegistry()
+        self.decimal = re.compile(r"\d+")
```
Comment (issy) [inline convert/convert.py:12]: let's move `decimal` to global and remove `__init__` altogether

File: convert/convert.py [hunk 2]
```diff
@@ -1,35 +1,66 @@
...
+        if " to " not in conversion:
+            await ctx.send(
+                f"`{conversion}` is not a valid conversion. Please make sure it is in the format `[p]convert <from> to <to>`"
+            )
+            await ctx.send_help()
+            return
+
+        arg1, end_unit = conversion.split(" to ")
+
+        amount = re.search(self.decimal, arg1).group()
```
Comment (issy) [inline convert/convert.py:32]:
```suggestion
        amount = self.decimal.search(arg1).group()
```
We can use search directly on the pattern

File: convert/convert.py [hunk 3]
```diff
@@ -1,35 +1,66 @@
...
+        except subprocess.CalledProcessError as e:
+            error = e.output.decode("utf-8")
+            # grab the first line for the error type
+            error_type = error.splitlines()[0]
+            embed = discord.Embed(
                 title="Error",
-                description=f"Unable to convert `{' '.join(unit)}`",
-                colour=await ctx.embed_colour(),
+                description=f"Error when converting `{conversion}`\n{error_type}",
+                color=0xFF0000,
```
Comment (issy) [inline convert/convert.py:48]:
```suggestion
                color=discord.Colour.red(),
```

File: convert/convert.py [hunk 4]
```diff
@@ -1,35 +1,66 @@
...
+            result = result.strip()
+            # check if first line has a * or /
+            if "*" in result or "/" in result:
+                # remove the first character
+                result = result[1:]
+            # create embed
+            embed = discord.Embed(
+                title="Convert",
+                description=f"`{conversion}`\n`{result.strip()}{end_unit}`",
+                color=0x00FF00,
```
Comment (issy) [inline convert/convert.py:63]:
```suggestion
                color=discord.Colour.green(),
```

**Review by issy — CHANGES_REQUESTED** [review]
Just a few minor suggestions. Looks good overall. If you could remove pint from the cog requirements (located in `convert/cog.json`) that would be much appreciated, and feel free to add yourself to the authors list if you want

**Comment by tigattack** [comment]
> feel free to add yourself to the authors list if you want

Good shout, there's also a list of contributors here that you're welcome to add yourself to https://github.com/rHomelab/LabBot-Cogs#other

**Comment by issy** [comment]
LGTM, thank you

**Review by issy — APPROVED** [review]

**Pull request merged**